### PR TITLE
fast track: Update main with missing commits

### PR DESCRIPTION
**Pull request by jamezp**
This PR is comprised of several commits that are in the release-3.2 and release-4.0 branches that should likely be in master. It also takes the liberty of change the master branches version from 3.1.0 to 4.0.0-SNAPSHOT. This was primarily done to prepare for a slimmed down 4.0.0 for Jakarta EE 11 as described in https://www.eclipse.org/lists/rest-dev/msg00231.html

Associated PR's that comprised this PR:

- #1211
- #1217
- #1220

Just for a reference too, this PR includes the following commits from release-4.0:

- #1182
- #1172
- #1145

### CODE REVIEW

**Comment by spericas** [comment]
So if master becomes the new 4.0 for EE11, isn't it confusing to have a 3.2 branch? I know that it is experimental, but seems a bit odd for its number to be lower. Maybe I'm missing something.

**Comment by jamezp** [comment]
> So if master becomes the new 4.0 for EE11, isn't it confusing to have a 3.2 branch? I know that it is experimental, but seems a bit odd for its number to be lower. Maybe I'm missing something.

TBH I completely agree. I'm just not sure what to do. The master branch is out of date. The release-4.0 branch can't really become master unless we revert some things like removing the the `@Context` and `@Suspended` annotations along with their respective documentation. The release-4.0 branch also has the TCK module commented out so it doesn't compile it and really it can't because of the removed annotations.

Both master, release-3.2 and release-4.0 include #1066 so we need some kind of version bump in master. If we remove the XML Binding dependency, then it needs to be a major bump.

In the end this seemed like the best middle ground. There really should be another commit on top of this, which I can add, that updates the signature test file too because it is missing the singantures from #1066 and the signature change from #863.

IMO none of the branches are in a releasable state. I will go ahead and add the commit to update the signature files to this PR too. I think that is critical too. Whether we go with this PR or not is fine, but IMO we do need some branch in a releasable state.

File: .github/workflows/maven-package.yml [hunk 1]
```diff
@@ -3,24 +3,24 @@ name: Jakarta REST CI
 
 on:
   push:
-    branches: [ master, 4.0-SNAPSHOT ]
+    branches: [ master ]
```
Comment (jim-krueger) [inline .github/workflows/maven-package.yml:6]: change to main

File: .github/workflows/maven-package.yml [hunk 2]
```diff
@@ -3,24 +3,24 @@ name: Jakarta REST CI
...
   pull_request:
-    branches: [ master, 4.0-SNAPSHOT ]
+    branches: [ master ]
```
Comment (jim-krueger) [inline .github/workflows/maven-package.yml:8]: change to main

**Comment by jim-krueger** [comment]
> Given, AIUI, this would take at least two weeks to get merged, that leaves with about two more weeks to get anything else done for Jakarta EE 11. We're due to have something by 29.Mar.2024, unless the schedule gets pushed.

I've added "fast track" to the title.  Since the changes in this PR have all been reviewed and approved in other PRs it seems logical to allow this to be fast tracked.

**Comment by jim-krueger** [comment]
> So if master becomes the new 4.0 for EE11, isn't it confusing to have a 3.2 branch?

I can rename the "release-3.2" branch to "release-4.1" to remove confusion.   If for some reason the EE11 schedule gets pushed out and the group feels comfortable with the content in the release-4.1 branch, I'm assuming we could apply it to 4.0.

File: .github/workflows/maven-package.yml (hunk 1 above)
Comment (jamezp) [reply .github/workflows/maven-package.yml]: Done

File: examples/src/main/java/jaxrs/examples/client/BasicExamples.java [hunk 3]
```diff
@@ -41,7 +41,6 @@
 import jakarta.ws.rs.ext.ReaderInterceptorContext;
 import jakarta.ws.rs.ext.WriterInterceptor;
 import jakarta.ws.rs.ext.WriterInterceptorContext;
```
Comment (jim-krueger) [inline examples/src/main/java/jaxrs/examples/client/BasicExamples.java:43]: Needs 2024 in copyright

File: examples/src/main/java/jaxrs/examples/client/spec/SpecExamples.java [hunk 4]
```diff
@@ -21,7 +21,6 @@
 import jakarta.ws.rs.core.Response;
 import static jakarta.ws.rs.client.Entity.entity;
```
Comment (jim-krueger) [inline examples/src/main/java/jaxrs/examples/client/spec/SpecExamples.java:23]: Needs 2024 in copyright

File: examples/src/main/java/jaxrs/examples/link/clusterservice/Cluster.java [hunk 5]
```diff
@@ -13,14 +13,11 @@
 import java.util.ArrayList;
 import java.util.List;
```
Comment (jim-krueger) [inline examples/src/main/java/jaxrs/examples/link/clusterservice/Cluster.java:15]: Add 2024

File: examples/src/main/java/jaxrs/examples/link/clusterservice/Machine.java [hunk 6]
```diff
@@ -10,14 +10,11 @@
 
 package jaxrs.examples.link.clusterservice;
```
Comment (jim-krueger) [inline examples/src/main/java/jaxrs/examples/link/clusterservice/Machine.java:11]: add 2024

File: jaxrs-spec/pom.xml [hunk 7]
```diff
@@ -27,13 +27,12 @@
     <parent>
         <groupId>jakarta.ws.rs</groupId>
         <artifactId>all</artifactId>
-        <version>3.1.0</version>
+        <version>4.0.0-SNAPSHOT</version>
     </parent>
 
     <properties>
         <site.output.dir>${project.build.directory}/staging</site.output.dir>
         <maven.site.skip>true</maven.site.skip>
```
Comment (jim-krueger) [inline jaxrs-spec/pom.xml:35]: add 2024 to copyright

File: jaxrs-spec/src/main/asciidoc/chapters/introduction/_status.adoc [hunk 8]
```diff
@@ -11,14 +11,14 @@
 [[status]]
```
Comment (jim-krueger) [inline jaxrs-spec/src/main/asciidoc/chapters/introduction/_status.adoc:11]: Add 2024 to copyright

File: jaxrs-tck-docs/userguide/pom.xml [hunk 9]
```diff
@@ -27,8 +27,8 @@
     <groupId>org.glassfish</groupId>
     <artifactId>tck_jaxrs</artifactId>
     <packaging>pom</packaging>
-    <version>3.1.2</version>
-    <name>Eclipse Foundation Technology Compatibility Kit User's Guide for Jakarta RESTful Web Services for Jakarta EE, Release 3.1</name>
+    <version>4.0.0-SNAPSHOT</version>
```
Comment (jim-krueger) [inline jaxrs-tck-docs/userguide/pom.xml:30]: Add 2024 to copyright

File: jaxrs-tck/pom.xml [hunk 10]
```diff
@@ -26,140 +26,19 @@
     <name>Jakarta RESTful WS TCK</name>
```
Comment (jim-krueger) [inline jaxrs-tck/pom.xml:26]: Add 2024 to copyright

File: jaxrs-tck/src/main/resources/ee/jakarta/tck/ws/rs/signaturetest/sig-test.map [hunk 11]
```diff
@@ -17,4 +17,4 @@
 ###############################################################
 # The signature test mapping file for the JAX-RS TCK.
 ###############################################################
-jakarta.ws.rs=3.1.0
+jakarta.ws.rs=4.0.0
```
Comment (jim-krueger) [inline jaxrs-tck/src/main/resources/ee/jakarta/tck/ws/rs/signaturetest/sig-test.map:20]: Add 2024 to copyright

File: pom.xml [hunk 12]
```diff
@@ -17,14 +17,14 @@
 
     <groupId>jakarta.ws.rs</groupId>
     <artifactId>all</artifactId>
-    <version>3.1.0</version>
+    <version>4.0.0-SNAPSHOT</version>
```
Comment (jim-krueger) [inline pom.xml:20]: Add 2024 to copyright

File: etc/config/checkstyle.xml [hunk 13]
```diff
@@ -92,7 +92,6 @@
         <!-- See http://checkstyle.sf.net/config_javadoc.html -->
 
         <module name="JavadocMethod">
```
Comment (jim-krueger) [inline etc/config/checkstyle.xml:94]: Add 2024 to copyright

File: examples/pom.xml [hunk 14]
```diff
@@ -22,198 +22,14 @@
     <parent>
         <groupId>jakarta.ws.rs</groupId>
         <artifactId>all</artifactId>
```
Comment (jim-krueger) [inline examples/pom.xml:24]: Add 2024 to copyright

**Comment by jamezp** [comment]
For ease of review I've done the copyright date updates in a separate commit. If preferred, I could do the updates within the correct commit the file was changed in.

**Review by jim-krueger — DISMISSED** [review]

**Review by jim-krueger — APPROVED** [review]

**Review by asoldano — APPROVED** [review]

**Review by jansupol — APPROVED** [review]

**Comment by jim-krueger** [comment]
Merging

**Pull request merged**